Directory.Build.props

```diff
@@ -134,7 +134,7 @@
     <!-- Default to all packages generating a corresponding symbol package -->
     <IncludeSymbols>true</IncludeSymbols>
     <!-- Workaround for AD0001 in analyzers with .NET 9. See https://github.com/dotnet/arcade/issues/14311 -->
-    <BuildWithNetFrameworkHostedCompiler>true</BuildWithNetFrameworkHostedCompiler>
+    <BuildWithNetFrameworkHostedCompiler Condition="'$(MSBuildRuntimeType)' == 'Full'">true</BuildWithNetFrameworkHostedCompiler>
```

Comment:
We shouldn't be setting this property in vstest anymore. That was only necessary for a short period of time during .NET 9. Please also update the comment above. GH doesn't let me do that.

```suggestion
```